### DISCUSSION TITLE
Load video from CDN does not work, possible to force the type of resource?

### DISCUSSION THREAD

**Discussion by naranjamecanica** [pixijs/pixijs#8588] [Q&A] [answered]
I'm loading video's from Vimeo's CDN (akamai) with `Texture.from()` which always worked, but in a new project it doesn't. The final CDN url for the movie is missing an `mp4` extension so the `autoDetectResource()` can't find out the resource type. 

Is there a way to force the resource type when using `Texture.from()` of `Texture.fromUrl()`?

The only way I can see to load them now is to mimic the behaviour defined in `Texture.from()` and create a `BaseTexture` myself...

Any thoughts? 👍🏻

**Comment by bigtimebuddy** [comment, marked as answer]
Texture.from can take an HTMLVideoElement as the first argument, so you can create and manage the video yourself.

**Reply by naranjamecanica** [reply]
Thanks for the fast response, I'll try that and will post an code example when it's working. 🙏🏻

**Comment by naranjamecanica** [comment]
Short working variant based on VideoResource (without listening to events, cleaning up, etc) is:

```typescript
// direct link to video
const videoCDNUrl = 'https://download-video.akamaized.net/playback/5953d08c....'

// video element
const videoElement = document.createElement('video');
videoElement.setAttribute('preload', 'auto');
videoElement.setAttribute('webkit-playsinline', '');
videoElement.setAttribute('playsinline', '');

// set crossOrigin
BaseImageResource.crossOrigin(videoElement, this.data.videoCDNUrl);

// add source to Video
const sourceElement = document.createElement('source');
sourceElement.src = videoCDNUrl;
sourceElement.type = `video/mp4`;
videoElement.append(sourceElement);

// create texture from video elemt
const videoTexture = Texture.from(videoElement, {
    resourceOptions: {
      updateFPS: 25, // video = 25fps
    },
});
```
